summarize/src/main.rs

```diff
@@ -0,0 +1,71 @@
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::time::Duration;
+use measureme::{ProfilingData, TimestampKind, Event};
+
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+struct Opt {
+    file_prefix: PathBuf,
+}
+
+fn main() {
+    let opt = Opt::from_args();
+
+    let data = ProfilingData::new(&opt.file_prefix);
+
+    let mut cumulative_time = HashMap::<_, Duration>::new();
+    let mut threads = HashMap::<_, Vec<Event>>::new();
+
+    for event in data.iter() {
+        match event.timestamp_kind {
+            TimestampKind::Start => {
+                let thread_stack = threads.entry(event.thread_id).or_default();
+
+                if let Some(prev_event) = thread_stack.last() {
+                    //count the time run so far for this event
+                    let duration =
+                        event.timestamp.duration_since(prev_event.timestamp)
+                            .unwrap_or(Duration::from_nanos(0));
+                    *cumulative_time.entry(prev_event.label.clone()).or_default() += duration;
```

Comment:
Not that it matters too much for a tool like this but cloning the `Cow<str>` potentially clones an entire `String` in each iteration here. It might be more performance to do something like:
```rust
if !cumulative_time.contains_key(&prev_event.label[..]) {
    cumulative_time.insert(prev_event.label.into_owned(), duration);
} else {
    cumulative_time[&prev_event.label[..]] += duration;
}
```
Not exactly elegant though `:)`

Actually the following would be better, as it just needs one lookup in the common case and doesn't clone the key:
```rust
if let Some(v) = cumulative_time.get_mut(&prev_event.label[..]) {
    *v += duration;
} else {
    cumulative_time.insert(prev_event.label.into_owned(), duration);
}
```